File: Agenda.py
class Agenda():
    def __init__(self, dow, start, end):
        if dow == 'U':
            days = 0
        elif dow == 'M':
            days = 1
        elif dow == 'T':
            days = 2
        elif dow == 'W':
            days = 3
        elif dow == 'R':
            days = 4
        elif dow == 'F':
            days = 5
        elif dow == 'S':
            days = 6
        print(start[0:2],'pls')
        self.start = (days*24*60+int(start[0:2])*60+int(start[2:]))
        self.end = (days*24*60+int(end[0:2])*60+int(end[2:]))
        '''
        start/end are minutes from 12:00 AM Sunday
        '''

File: util/Agenda.js
class Agenda {
    constructor(dow, start, end) {
        var days = 0
        if (dow == 'U') {
            days = 0
        } else if (dow == 'M') {
            days = 1 
        } else if (dow == 'T') {
            days = 2
        } else if (dow == 'W') {
            days = 3
        } else if (dow == 'R') {
            days = 4
        } else if (dow == 'F') {
            days = 5
        } else if (dow == 'S') {
            days = 6
        }
        this.start = (days*24*60+parseInt(start.substring(0,2),10)*60+parseInt(start.substring(2),10))
        this.end = (days*24*60+parseInt(end.substring(0,2),10)*60+parseInt(end.substring(2),10))
        this.timeRange = d3.range(this.start,this.end)
        // start/end are minutes from 12:00 AM Sunday
    }

    equals(o) {
        return o instanceof Agenda && this.timeRange == o.timeRange
    }

    toString() {
        return "(" + this.start.toString() + "," + this.end.toString() + ")"
    }
}
